lib/services/wallet_connect_service.dart

```diff
@@ -0,0 +1,355 @@
+    web3Wallet!.registerEventEmitter(
+        chainId: Config.walletConnectChainId,
+        event: WcEvents.tokenAmountChanged);
+    web3Wallet!.registerEventEmitter(
+        chainId: Config.walletConnectChainId, event: WcEvents.accountsChanged);
+    web3Wallet!.registerEventEmitter(
+        chainId: Config.walletConnectChainId, event: WcEvents.methodResult);
+
+    // -------------------------------------------------------- METHODS --------------------------------------------------------
+    // requestAccounts (responds automatically to clients)
+    web3Wallet!.registerRequestHandler(
+        chainId: Config.walletConnectChainId,
+        method: WcMethods.wRequestAccounts,
+        handler: (topic, args) {
+          List<dynamic> data = [];
+          appStore.currentQubicIDs.forEach(((id) {
+            if (id.watchOnly == false) {
+              dynamic item = {};
+              item["address"] = id.publicId;
+              item["name"] = id.name;
+              item["amount"] = id.amount ?? -1;
+              data.add(item);
+            }
+          }));
+          web3Wallet!.emitSessionEvent(
+              topic: topic,
+              chainId: Config.walletConnectChainId,
+              event: SessionEventParams(
+                  name: WcEvents.accountsChanged, data: data));
+        });
+
+    // sendQubic emits an onRequestSendQubic event if the request is valid
+    // otherwise returns a validation error
+    web3Wallet!.registerRequestHandler(
+        chainId: Config.walletConnectChainId,
+        method: WcMethods.wSendQubic,
+        handler: (topic, args) {
+          late RequestSendQubicEvent result;
+          try {
+            result = RequestSendQubicEvent.fromMap(
+                args, topic, args["nonce"].toString());
+            result.validateOrThrow();
+
+            //Enhance the result with the pairing metadata
+            if (web3Wallet!.getActiveSessions().containsKey(topic)) {
+              result.setPairingMetadata(
+                  web3Wallet!.getActiveSessions()[topic]!.peer.metadata);
+              onRequestSendQubic.add(result);
+            }
+          } catch (e) {
+            emitErrorSessionEvent(
+                topic, e.toString(), args["nonce"].toString());
+          }
+        });
+
+    web3Wallet!.registerRequestHandler(
+        chainId: Config.walletConnectChainId,
+        method: WcMethods.wSendAsset,
+        handler: (name, args) {});
+
+    // -------------------------------------------------------- END OF METHODS ---------------------------------------------------------
+    web3Wallet!.registerAccount(
+        accountAddress:
+            "000000000000000000000000000000000000000000000000000000000000",
+        chainId: Config.walletConnectChainId);
+
+    appStore.currentQubicIDs.forEach(((id) {
+      if (id.watchOnly == false) {
+        web3Wallet!.registerAccount(
+            accountAddress: id.publicId, chainId: Config.walletConnectChainId);
```

Comment:
According to the method documentation:
/// Each account must follow the namespace:chainId:address format or this will throw an error.

Reply:
Yes, the registerAccount automatically generated the CAIP10 format when needed.